src/main/java/com/assignment/application/controller/SalaryController.java

```diff
@@ -14,30 +18,93 @@
-        return salaryServiceI.getSalary(companyId,empId);
+                                                @PathVariable("emp_id") String employeeId,
+                                                @RequestHeader("username") String username,
+                                                @RequestHeader("password") String password) {
+        try {
+            if (verifyUser.authorizeUser(username, password) == 1 || (verifyUser.authorizeEmployee(username, password) == 1 && employeeId.equalsIgnoreCase(username))) {
+                Salary salary = salaryServiceI.getSalary(companyId, employeeId);
+                if (salary == null) {
+                    return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+                }
+                return new ResponseEntity<>(salary, HttpStatus.OK);
+            }
+            return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+        } catch (Exception e) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
     }
 
     @GetMapping(value = "/salary")
-    public ResponseEntity<List<Salary>> getSalaryList(){
-        return salaryServiceI.getSalaryList();
+    public ResponseEntity<List<Salary>> getSalaryList(@RequestHeader("username") String username,
+                                                      @RequestHeader("password") String password) {
+        try {
+            int status = verifyUser.authorizeUser(username, password);
+            if (status == 0) {
+                return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+            }
+            List<Salary> salaryList = salaryServiceI.getSalaryList();
+            if (salaryList == null) {
+                return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+            }
+            return new ResponseEntity<>(salaryList, HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
     }
 
     @PatchMapping(value = "{comp_id}/salary-update")
     public ResponseEntity<String> updateSalary(@PathVariable("comp_id") Long companyId,
-                                               @RequestBody SalaryUpdate salaryUpdate){
-        return salaryServiceI.updateSalary(companyId,salaryUpdate);
+                                               @RequestBody SalaryUpdate salaryUpdate,
+                                               @RequestHeader("username") String username,
+                                               @RequestHeader("password") String password) {
+        try {
+            int status = verifyUser.authorizeUser(username, password);
+            if (status == 0) {
+                return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+            }
+            if (salaryServiceI.updateSalary(companyId, salaryUpdate).equalsIgnoreCase(StringConstant.UPDATE_SUCCESSFUL)) {
+                kafkaTemplateSalary.send(TOPIC, salaryUpdate);
```

Comment:
I would love to see kafka triggers in service layer rather than controller layer to keep it view specific